src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -57,8 +58,8 @@ pub(super) struct LocationState {
 impl LocationState {
     /// Default initial state has never been accessed and has been subjected to no
     /// foreign access.
-    fn new(permission: Permission) -> Self {
-        Self { permission, initialized: false, latest_foreign_access: None }
+    fn new(initialized: bool, permission: Permission) -> Self {
+        Self { permission, initialized, latest_foreign_access: None }
```

Comment:
May be worth a `debug_assert!` to ensure that if `permission` is `Active` then `initialized` is `true`. After all we rely on this in the proof...

Reply:
I've added a debug assert.